### PR TITLE
Public docs for jparse(), mapKeys(), and mapValues()

### PR DESCRIPTION
**Pull request by amohammed-2**
This PR provides documentation for the new `jparse()`, `mapKeys()`, and `mapValues()` functions which will be going into GA.

Jira Issue: NR-181555

### CODE REVIEW

**Comment by bradleycamacho** [comment]
Thanks for the pr @amohammed-2! I'm moving this to our drafts column for now. Once it's ready for review, please ping the hero in #help-documentation. Thanks!

File: src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx [hunk 1]
```diff
@@ -2841,6 +2841,209 @@ Note: `aparse()` is case-insensitive.
     </CollapserGroup>
   </Collapser>
 
+  <Collapser
```
Comment (mlobiondo-newrelic) [inline src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx:2844]: This should be in alphabetical order between 'if' and 'length'.

File: src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx [hunk 2]
```diff
@@ -2841,6 +2841,209 @@ Note: `aparse()` is case-insensitive.
...
+    className="freq-link"
+    id="func-jparse"
+    title={<InlineCode>jparse(attribute [, path])</InlineCode>}
+  >
+    Use the JSON parse function, `jparse()`, to parse a string value and produce a Map / List of values (or nested structures) which can be handled like any other first-class value type in NRQL.
```
Comment (mlobiondo-newrelic) [inline src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx:2849]: We don't generally capitalize data types.

File: src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx [hunk 3]
```diff
@@ -2841,6 +2841,209 @@ Note: `aparse()` is case-insensitive.
...
+
+    `jparse()` takes two arguments:
+    * attribute - A JSON string value. 
+    * path - An optional string value that is used to directly reference a particular piece of the JSON within the `attribute` parameter. See the [JSON Parse Path Syntax Reference](#json-parse-path-syntax-reference) section below.
+
+
+    The `jparse()` function follows the [RFC 8259 format](https://datatracker.ietf.org/doc/html/rfc8259#section-2) to parse JSON values. When the `jparse()` function is used without the `path` parameter, it will return the deserialized JSON value.
+    
+
+    You can use the square-bracket syntax to pull out individual values from a `jparse()` result via a key/index, and map JSON keys directly to attributes using the `WITH` clause.
```
Comment (mlobiondo-newrelic) [inline src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx:2859]: "You can use square brackets" is less wordy and I believe conveys the same information.

File: src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx [hunk 4]
```diff
@@ -2841,6 +2841,209 @@ Note: `aparse()` is case-insensitive.
...
+       **Referencing an index**
+        
+        The following query references index `0` within the `jsonString` attribute and will return `'abc'`.
+
+        ```sql
+        WITH '["abc", "xyz"]' as jsonString SELECT jparse(jsonString)[0]
+        ```
+      </Collapser>
+      <Collapser id="mapping-json-keys-to-attributes" title="Mapping JSON keys to attributes">
+        The following query uses `jparse()` in the `WITH` clause to map the JSON keys `userName` and `id` directly into attributes so they can be used in the rest of the query.
```
Comment (mlobiondo-newrelic) [inline src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx:2880]: ...to map the JSON keys `userName` and `id` into NRQL variables so...

File: src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx [hunk 5]
```diff
@@ -2841,6 +2841,209 @@ Note: `aparse()` is case-insensitive.
...
+          WITH '{"user": {"name": "John", "id": 5}}' as jsonString SELECT jparse(jsonString)
+        ```
+
+        This query will return the deserialized JSON string:
+        ```
+          {"user":{"name":"John","id":5}}
+        ```
+      </Collapser>
+      <Collapser id="parse-specific-value-from-log-message" title="Parse a specific value from a log message">
+        A common problem is having rich and structured data hiding with a Log message. You can leverage `aparse()` and `jparse()` to trim away noise and find specific values within a log message.
```
Comment (mlobiondo-newrelic) [inline src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx:3022]: ...rich and structured data within a Log message...?

File: src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx (hunk 1 above)
Comment (amohammed-2) [reply src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx]: Fixed.

File: src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx [hunk 6]
```diff
@@ -2895,6 +3098,50 @@ Note: `aparse()` is case-insensitive.
     _Related function: [<InlineCode>upper()</InlineCode>](#func-upper)_
   </Collapser>
 
+  <Collapser
+    className="freq-link"
+    id="func-mapKeys"
+    title={<InlineCode>mapKeys(attribute)</InlineCode>}
+  >
+    Use the `mapKeys()` function to return a list of keys when provided a Map as input within the `attribute` parameter.
```
[the remark this answers is not in the document]
Comment (amohammed-2) [reply src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx]: Removed capitalization from data types.

File: src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx (hunk 2 above)
Comment (amohammed-2) [reply src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx]: Fixed.

File: src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx (hunk 3 above)
Comment (amohammed-2) [reply src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx]: Sounds good, updated the text accordingly.

File: src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx (hunk 4 above)
Comment (amohammed-2) [reply src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx]: Fixed.

File: src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx (hunk 5 above)
Comment (amohammed-2) [reply src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx]: Fixed.

**Review by jeff-colucci — APPROVED** [review]
Looks good!

**Comment by jeff-colucci** [comment]
Gatsby built successfully after a cache clear and rebuild, so I'm publishing.

**Pull request merged**